src/App.js

```diff
@@ -1,26 +1,49 @@
 import React from 'react';
-import logo from './logo.svg';
 import './App.css';
+import { firebase, db } from './lib/firebase';
+import { useCollection } from 'react-firebase-hooks/firestore';
+
+const App = () => {
+  const [value, loading, error] = useCollection(
+    firebase.firestore().collection('list'),
+    {
+      snapshotListenOptions: { includeMetadataChanges: true },
+    },
+  );
+
+  const sendItem = () => {
+    firebase
+      .firestore()
+      .collection('list')
+      .add({ title: 'first item', description: 'new item' })
+      .then((documentReference) => {
+        console.log('document reference ID', documentReference.id);
+      })
+      .catch((error) => {
+        console.log(error.message);
+      });
+  };
 
-function App() {
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div>
+      <h1>Grocery List</h1>
+      <button onClick={sendItem}>click here to add grocery item</button>
+      <p>
+        {error && <strong>Error: {JSON.stringify(error)}</strong>}
+        {loading && <span>Collection: Loading...</span>}
+        {value && (
+          <span>
+            Collection:{' '}
+            {value.docs.map((doc) => (
+              <React.Fragment key={doc.id}>
+                {JSON.stringify(doc.data())},{' '}
+              </React.Fragment>
+            ))}
+          </span>
+        )}
```

Comment:
|before|after|
|---|---|
|![firebase-render-before](https://user-images.githubusercontent.com/47455758/114286237-925bfe00-9a22-11eb-8bf8-3a1825483e1b.jpg)|![firebase-render-refactor](https://user-images.githubusercontent.com/47455758/114286241-9ab43900-9a22-11eb-8a93-75ed7ae70eb5.jpg)|

### Proposed Changes:

1. I thought we could refactor the App.js return section featuring `<h1>Grocery List </h1>`  with a `<ul>`  and `<li>` instead of a `<span>`
2. I got rid of the `React.Fragment` and replaced it with the newer `<>` and `</>` syntax, wrapping the following block: 
```javascript
<>
    <p>Collection:</p>
    <ul>
        {value.docs.map((doc) => (
            <li key={doc.id}>{JSON.stringify(doc.data())} </li>
         ))}
    </ul>
</>
```
3. I replaced the `<strong>` on line 32 with a `<h2>` and the `<span>` on line 33 with a `<p>` to try and maintain good semantic HTML practices.

### UI

![grocery-list-li-render](https://user-images.githubusercontent.com/47455758/114286496-cf28f480-9a24-11eb-8e20-049ec0345617.jpg)

### Conclusion

- I think this gives us a more visually palatable representation of the data, and gets us closer to best HTML practices.
- I really like how you handle loading data and error messages, which looks great in both code and real time!

Reply:
Great ideas, @jamesncox! This will prepare nicely for some of the upcoming issues!

Reply:
James - I like it all of that! I'll speak for myself in that I got so focused on getting all things Firebase to work that the visual layout and semantics were not even on my mind, but those are very helpful steps to implement that set the upcoming stories up with a little less to fine tune.